src/api/apiService.ts

```diff
@@ -36,6 +36,23 @@ export async function fetchDefaultNamespace() {
   }
 }
 
+export async function fetchStepDetails(id: string | undefined, namespace?: string) {
```

Comment:
for this we can do `id?: string` instead of `string | undefined`, just like with `namespace`